Sources/DataPipeline/Config/SDKConfigBuilder.swift

```diff
@@ -0,0 +1,147 @@
+import CioInternalCommon
+import Foundation
+import Segment
+
+/// Builder class designed to facilitate the creation of `SdkConfig` and `DataPipelineConfigOptions`.
+/// The builder pattern allows for a fluent and intuitive way to set up configuration options for
+/// the SDK, ensuring all required and optional settings are appropriately configured before the
+/// SDK is initialized.
+///
+/// Usage Example:
+/// ```
+/// let config = SDKConfigBuilder(writeKey: "your_write_key")
+///   .apiHost("your_api_host")
+///   .flushAt(30)
+///   // additional configuration as needed...
+///   .build()
+/// // Use `config` for initializing the SDK...
+/// ```
+public class SDKConfigBuilder {
+    // default static values for configuration options
+    private static let defaultAPIHost = "cdp.customer.io/v1"
+    private static let defaultCDNHost = "cdp.customer.io/v1"
+
+    // configuration options for SdkConfig
+    private var logLevel: CioLogLevel = .error
+
+    // configuration options for DataPipelineConfigOptions
+    private let writeKey: String
+    private var apiHost: String = SDKConfigBuilder.defaultAPIHost
+    private var cdnHost: String = SDKConfigBuilder.defaultCDNHost
+    private var flushAt: Int = 20
+    private var flushInterval: Seconds = 30
+    private var autoAddCustomerIODestination: Bool = true
+    private var defaultSettings: Settings?
+    private var flushPolicies: [FlushPolicy] = [CountBasedFlushPolicy(), IntervalBasedFlushPolicy()]
+    private var flushQueue: DispatchQueue = .init(label: "com.segment.operatingModeQueue", qos: .utility)
```

Comment:
Instead of having hard-coded values here in our SDK that match the Segment default, could we reference the Segment default values? 

```suggestion
    private var flushPolicies: [FlushPolicy] = SegmentSdk.defaultFlushPolicy
```

Reply:
Unfortunately, Segment also [hardcodes default values](https://github.com/customerio/cdp-analytics-swift/blob/main/Sources/Segment/Configuration.swift) in a similar fashion. So, unless we make changes to the Segment SDK, there's no direct way to reference these values.

I'm currently writing tests to ensure that our values match Segment's, will open a PR for it soon.